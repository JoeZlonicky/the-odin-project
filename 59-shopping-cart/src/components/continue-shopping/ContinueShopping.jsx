import { NavLink } from 'react-router-dom';
import styles from './ContinueShopping.module.css';

function ContinueShopping() {
  return (
    <NavLink to="/all-products" className={styles.continueShopping}>
      Continue Shopping
    </NavLink>
  );
}

export default ContinueShopping;
