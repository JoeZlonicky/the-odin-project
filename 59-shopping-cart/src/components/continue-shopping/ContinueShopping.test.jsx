import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { expect, it } from 'vitest';
import ContinueShopping from './ContinueShopping';

it('Matches screenshot', () => {
  const { container } = render(
    <MemoryRouter>
      <ContinueShopping />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});

it('Correctly navigates to all products', async () => {
  const user = userEvent.setup();

  render(
    <MemoryRouter>
      <Routes>
        <Route path="/" element={<ContinueShopping />} />
        <Route path="/all-products" element={<h1>All Products</h1>} />
      </Routes>
    </MemoryRouter>,
  );
  await user.click(screen.getByText('Continue Shopping'));

  const header = screen.getByRole('heading', { level: 1 });
  expect(header.textContent).toMatch('All Products');
});
