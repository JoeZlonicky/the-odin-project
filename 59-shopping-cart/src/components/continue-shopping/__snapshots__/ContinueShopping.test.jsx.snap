// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches screenshot 1`] = `
<div>
  <a
    class="_continueShopping_8c1960"
    href="/all-products"
  >
    Continue Shopping
  </a>
</div>
`;
